modules/vba_documents/app/sidekiq/vba_documents/upload_remover.rb

```diff
@@ -21,21 +21,20 @@ class UploadRemover
     def perform
       return unless Settings.vba_documents.s3.enabled
 
-      rows = VBADocuments::UploadSubmission.select(:id, :guid, :status, :s3_deleted, :created_at)
-                                           .where(REMOVAL_QUERY, EXPIRATION_TIME.ago)
+      exp_time = EXPIRATION_TIME.ago
 
-      rows.find_in_batches do |batch|
-        VBADocuments::UploadSubmission.transaction do
-          batch.each do |upload|
-            Rails.logger.info("VBADocuments: Cleaning up s3: #{upload.inspect}")
+      loop do
```

Comment:
I know an infinite loop looks weird, but it's pretty much logically the same as the find_in_batches it replaced.
In either case, the "algo" just loops until no more records are found or the sidekick job itself is terminated(like due to a server recycle).  If the job does get interrupted, it's idempotent, the next run in 5 minute will picked up the missed records 

If the query does not get stuck in postgres waiting or disk reads(which has been addressed with hardware upgrade), the job is actually pretty efficient, most of the time the entire job run process less than 100 records, so 4 loops, and it completes in just a few seconds.

Reply:
Would it make sense to add a MAX_ITERATIONS and break if reached? Say 10k or 100k?

I could also see not doing a loop at all and instead just enqueuing another `UploadRemover` job if there's still more submissions to remove.

```ruby
class VBADocuments
  class UploadRemover
    # ...

    def perform
      return unless Settings.vba_documents.s3.enabled

      process_batch(EXPIRATION_TIME.ago)
    end

    private

    def process_batch(exp_time)
      submission_guids = fetch_batch(exp_time)
      return if submission_guids.empty?

      delete_s3_objects(submission_guids)
      mark_submissions_deleted(submission_guids)

      self.class.perform_async if more_submissions_exist?(exp_time)
    end
    
    # ...
  end
end
```